In [1]:
import pandas as pd

In [3]:
df=pd.read_csv('FoodBalanceSheets_E_Africa_NOFLAG.csv',encoding='Latin')

In [4]:
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

>  Question 1:What is the total sum of Animal Fat produced in 2014 and 2017 respectively?

In [9]:
df['Item'].value_counts().head(15)

Milk - Excluding Butter       1262
Eggs                          1142
Miscellaneous                  911
Cereals - Excluding Beer       714
Oilcrops                       706
Maize and products             687
Starchy Roots                  674
Rice and products              670
Wheat and products             653
Pulses                         649
Potatoes and products          641
Animal fats                    639
Pulses, Other and products     626
Groundnuts (Shelled Eq)        617
Sugar & Sweeteners             613
Name: Item, dtype: int64

In [14]:
df_animal=df[df['Item']=='Animal fats']
df_animal.sum()

Area Code                                                   85279
Area            AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...
Item Code                                                 1882494
Item            Animal fatsAnimal fatsAnimal fatsAnimal fatsAn...
Element Code                                              2567303
Element         ProductionImport QuantityStock VariationExport...
Unit            1000 tonnes1000 tonnes1000 tonnes1000 tonnes10...
Y2014                                                   209460.54
Y2015                                                   200675.72
Y2016                                                   183314.22
Y2017                                                   269617.53
Y2018                                                   269648.27
dtype: object

- 209460.54 and 269617.53

>  Question 2: What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [15]:
df['Y2015'].describe()

count     59395.000000
mean        135.235966
std        1603.403984
min       -3161.000000
25%           0.000000
50%           0.080000
75%           8.460000
max      181137.000000
Name: Y2015, dtype: float64

- 135.236 and 1603.404

>  Question 3: What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [27]:
df['Y2016'].isna().sum()

1535

In [26]:
(df['Y2016'].isna().sum())/(df['Y2016'].size)*100

2.5187470259094566


- 1535 and 2.52

>  Question 4: 
Which year had the highest correlation with ‘Element Code’? 

In [30]:
df.corr()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

- 2014

>  Question 5: What year has the highest sum of Import Quantity?

In [31]:
df_import=df[df['Element']=='Import Quantity']
df_import.sum()

Area Code                                                  688174
Area            AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...
Item Code                                                13795966
Item            Cereals - Excluding BeerWheat and productsRice...
Element Code                                             28834929
Element         Import QuantityImport QuantityImport QuantityI...
Unit            1000 tonnes1000 tonnes1000 tonnes1000 tonnes10...
Y2014                                                   274144.48
Y2015                                                   267018.46
Y2016                                                   286582.78
Y2017                                                   294559.09
Y2018                                                   287997.09
dtype: object

- 2017

> Question 6:What is the total number of the sum of Production in 2014?

In [36]:
df.groupby('Element')['Y2014'].sum().sort_values()

Element
Tourist consumption                           416.00
Fat supply quantity (g/capita/day)          10225.56
Protein supply quantity (g/capita/day)      11836.46
Seed                                        21922.92
Residuals                                   30149.00
Food supply quantity (kg/capita/yr)         49650.63
Stock Variation                             58749.83
Other uses (non-food)                       78718.13
Export Quantity                            150020.64
Losses                                     153223.00
Feed                                       216927.89
Import Quantity                            274144.48
Processing                                 282923.00
Food supply (kcal/capita/day)              454257.00
Total Population - Both sexes             1031585.00
Food                                      1212332.49
Production                                1931287.75
Domestic supply quantity                  1996716.35
Name: Y2014, dtype: float64

- 1931287.75

> Question 7:
Which of these elements had the highest sum in 2018?

In [37]:
df.groupby('Element')['Y2018'].sum().sort_values()

Element
Tourist consumption                            90.00
Fat supply quantity (g/capita/day)          10258.69
Protein supply quantity (g/capita/day)      11833.56
Stock Variation                             20577.91
Seed                                        25263.14
Residuals                                   34864.00
Food supply quantity (kg/capita/yr)         49056.85
Other uses (non-food)                       91300.97
Losses                                     163902.00
Export Quantity                            181594.80
Feed                                       233489.68
Import Quantity                            287997.09
Processing                                 308429.00
Food supply (kcal/capita/day)              455261.00
Total Population - Both sexes             1140605.00
Food                                      1303841.28
Production                                2075072.89
Domestic supply quantity                  2161192.10
Name: Y2018, dtype: float64

- Domestic supply quantity   

> Question 8: Which of these elements had the 3rd lowest sum in 2018?

In [40]:
df.groupby('Element')['Y2018'].sum().sort_values()

Element
Tourist consumption                            90.00
Fat supply quantity (g/capita/day)          10258.69
Protein supply quantity (g/capita/day)      11833.56
Stock Variation                             20577.91
Seed                                        25263.14
Residuals                                   34864.00
Food supply quantity (kg/capita/yr)         49056.85
Other uses (non-food)                       91300.97
Losses                                     163902.00
Export Quantity                            181594.80
Feed                                       233489.68
Import Quantity                            287997.09
Processing                                 308429.00
Food supply (kcal/capita/day)              455261.00
Total Population - Both sexes             1140605.00
Food                                      1303841.28
Production                                2075072.89
Domestic supply quantity                  2161192.10
Name: Y2018, dtype: float64

- Protein supply quantity (g/capita/day)

> Question 9:
What is the total Import Quantity in Algeria in 2018?

In [50]:
df_import=df[df['Element']=='Import Quantity']

In [54]:
df_import

Area Code     Area  Item Code                      Item  Element Code  \
12             4  Algeria       2905  Cereals - Excluding Beer          5611   
28             4  Algeria       2511        Wheat and products          5611   
44             4  Algeria       2805         Rice and products          5611   
59             4  Algeria       2513       Barley and products          5611   
74             4  Algeria       2514        Maize and products          5611   
...          ...      ...        ...                       ...           ...   
60897        181  Algeria       2769   Aquatic Animals, Others          5611   
60905        181  Algeria       2775            Aquatic Plants          5611   
60913        181  Algeria       2928             Miscellaneous          5611   
60923        181  Algeria       2680               Infant food          5611   
60933        181  Algeria       2899             Miscellaneous          5611   

               Element         Unit     Y2014    Y2015    Y2016    Y2017  \
12     Import Quantity  1000 tonnes  12604.00  13983.0  13523.0  13039.0   
28     Import Quantity  1000 tonnes   7461.00   8531.0   8255.0   8105.0   
44     Import Quantity  1000 tonnes    171.00    164.0    162.0    190.0   
59     Import Quantity  1000 tonnes    805.00    797.0    916.0    567.0   
74     Import Quantity  1000 tonnes   4148.00   4450.0   4159.0   4161.0   
...                ...          ...       ...      ...      ...      ...   
60897  Import Quantity  1000 tonnes      0.01      0.0      0.0      0.0   
60905  Import Quantity  1000 tonnes      0.00      0.0      0.0      0.0   
60913  Import Quantity  1000 tonnes     42.00     47.0     34.0     15.0   
60923  Import Quantity  1000 tonnes      0.00      0.0      0.0      0.0   
60933  Import Quantity  1000 tonnes     42.00     47.0     33.0     15.0   

         Y2018  
12     13182.0  
28      8434.0  
44       207.0  
59       402.0  
74      4126.0  
...        ...  
60897      0.0  
60905      0.0  
60913     16.0  
60923      0.0  
60933     16.0  

[5139 rows x 12 columns]

In [56]:
df_import_area=df_import[df_import['Area']=='Algeria']

In [61]:
df_import_area["Y2018"].sum()

287997.08999999997

- 287997.09

> Question 10:
What is the total number of unique countries in the dataset?

In [63]:
df['Area'].nunique()

49

-  49